File: Java/Invert Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode invertTree(TreeNode root) {
        if (root == null) return null;
        swap(root);
        return root;
    }
    
    private void swap(TreeNode root) {
        TreeNode tmp = root.left;
        root.left = root.right;
        root.right = tmp;
        
        if (root.left != null)
            swap(root.left);
        if (root.right != null)
            swap(root.right);
    }
}

File: Python/Invert Binary Tree .py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    """
    :type root: TreeNode
    :rtype: TreeNode
    """
    def invertTree(self, root):
        if root == None:
            return root
        self.swap(root)
        return root
    
    def swap(self, root):
        tmp = root.left
        root.left = root.right
        root.right = tmp
        
        if root.left != None:
            self.swap(root.left)
        if root.right != None:
            self.swap(root.right)